### ISSUE TITLE
Ladepunkte verdecken einander

### ISSUE THREAD

**Issue by secuder** [remuslazar/EVPlugFinderPublic#63]
Keine Ahnung, wie oft das vorkommt, aber hier verdeckt ein Ladepunkte den anderen. Erst beim Zoomen (und hier zufällig) sah ich, das das zwei verschiedene Ladepunkte sind. 

Wäre schön, wenn die Ladepunkte - zumindest wenn es außen rum so leer ist wie hier - beide sichtbar wären. Oder dass zumindest der schnellere Ladepunkte "oben" angezeigt wird.

**Comment by secuder** [comment]
![img_4152](https://cloud.githubusercontent.com/assets/24484266/25044756/b3cd764e-2128-11e7-82ac-dc5483a1bd9f.PNG)
![img_4153](https://cloud.githubusercontent.com/assets/24484266/25044755/b3cc1920-2128-11e7-932b-fe409abefba3.PNG)

**Comment by remuslazar** [comment]
@secuder ja, das ist mir auch schon mal passiert, ich denke, das ist äußerst selten, dass man so eine Situation hat. Was jedenfalls hilft, zum. als Workaround erst mal, den "Schnelllade-Filter" zu benutzen, wenn man sowieso "unterwegs" was sucht zum schnell laden. Dadurch kann man solche Situationen auflösen.

Um das sauber zu lösen müsste man es clustern. Algorithmen und auch eine Swift-Library gibt es schon, die Umsetzung ist jedoch nicht ganz trivial, weil ich nicht dauernd neue Requests machen will, wenn man die Karte zoomt, deswegen werde ich das erst mal zurückstellen.

**Comment by secuder** [comment]
Allgemeines Clustern ist im Forum ja durchgefallen ;-) so dass ein Clustern nur dann sinnvoll ist, wenn sich die Ladepunkte tatsächlich überdecken - abhängig von dem jeweils gewählten Kartenausschnitt / des jeweiligen Maßstabs. Müsste lösbar sein, aber sicher nicht so ohne weiteres...

**Comment by remuslazar** [comment]
Das Clustering habe ich noch als PR offen. Ich bin selber noch nicht ganz zufrieden mit dem Ergebnis, habe aber schon einige coole Ideen, wo aber eine Implementierung ziemlich anspruchsvoll wird, ich habe das schon vor Kurzem im Slack beschrieben. Wenn das kommt, wird es indirekt auch dieses Problem hier lösen.

Bis dahin könnte man mit einem Clustering Algorithmus benachbarte Stationen finden. Allerdings ist dann der Aufwand O(n^2). Da man diese Berechnung nur einmalig nach dem Fetch machen muss, wäre das noch zu verkraften. Ist dann allerdings immer noch offen, wie man das visuell anders darstellt, da habe ich ad hoc auch nicht direkt eine gute Lösung parat.

Daher würde ich vorschlagen, bis auf weiteres einfach mit dem passenden Filter arbeiten. Wenn man ja sowieso nur Schnelllader will, so kann man gut mit dem Filter arbeiten, dann ist die „darunter liegende“ Station gar nicht mehr in der Ergebnismenge enthalten und stört nicht mehr.

**Comment by secuder** [comment]
Siehst Du die Möglichkeit, die "schnelle" Farbe nach oben zu holen? Das wäre vielleicht leichter machbar und in jedem Falle eine Verbesserung.

**Comment by remuslazar** [comment]
Priorität und vieles mehr kann man mit iOS11 machen. Ich lasse das offen, werde ich aber erst angehen, wenn iOS11 offiziell verfügbar ist.
